src/nvim/terminal.c

```diff
@@ -295,7 +295,10 @@ void terminal_close(Terminal *term, char *msg)
   }
 
   if (buf) {
```

Comment:
```suggestion
  if (buf && !exiting)) {
```

Reply:
> we could instead check is_autocmd_blocked(), which is probably more robust but perhaps doesn't express the intention.

Is there ever a scenario where the `v:` dict is already freed but autocommands are *not* blocked (I assume not but I want to be sure)?

Reply:
I think I lean toward using `!exiting`. As you say I think the intention is clearer, and `is_autocmd_blocked()` is already checked in the call to `apply_autocmds`.

Reply:
> Is there ever a scenario where the `v:` dict is already freed but autocommands are _not_ blocked (I assume not but want to be sure)?

No. I just like the semantics of checking `exiting`. But it's more correct to check `is_autocmd_blocked()` here, because `exiting`.